src/test/cli.build.test.ts

```diff
@@ -27,6 +27,21 @@ describe('Dev Containers CLI', function () {
 
 	describe('Command build', () => {
 
+		it('should correctly configure the image name to push from --image-name with --push true', async () => {
+			let success = false;
+			const testFolder = `${__dirname}/configs/example`;
+			try {
+				await shellExec(`${cli} build --workspace-folder ${testFolder} --image-name demo:v1 --push true`);
+			} catch (error) {
+				assert.equal(error.error.code, 1, 'Should fail with exit code 1');
+				const res = JSON.parse(error.stdout);
+				assert.equal(res.outcome, 'error');
+				assert.match(JSON.stringify(error.error.message), /unauthorized: access token has insufficient scopes/);
+				assert.match(JSON.stringify(error.error.message), /pushing demo:v1 with docker/);
+			}
+			assert.equal(success, false, 'expect non-successful call');
+		});
```

Comment:
Could you add check looking up all tags on `demo:v1` and asserting there are no additional tags? Maybe that could even replace the above string matches which might break with string changes to the upstream Docker CLI.

Reply:
@prathameshzarkar9 Christof is referring to using something like 👇 to confirm that the image does not refer to any additional tags except `demo:v1` . Once you add this code in, you can remove the string assertions as it cam break in future

```
tags=$(docker images --format "{{.Tag}}" demo:v1)
```

Reply:
please check the new changes committed to have unique images list